include/Network/ENetPacketIStream.h

```diff
@@ -39,6 +39,8 @@ class ENetPacketIStream final : public InputStream {
         }
     }
 
+    long bytesLeft() const override { return packet->dataLength - currentPos; }
```

Comment:
Can serialized data cross packet boundaries?  If so, we might want a sentinel value (e.g., "-1") or some-such to indicate an unknown stream size that would fall back to the old behavior (read until exception).

Reply:
> Can serialized data cross packet boundaries? 

My impression is that it would not.

Even if it does, the behavior should be identical here to how it worked before, with the only exception that line https://github.com/henricj/dunelegacy/pull/34/files#diff-1ec625d8f6a9012cde14119737e7a5b614f5ec6b750d954519ac794f2792fdb1R67 will now print a newly added warning message.

So if there is data serialization, then that warning message will begin to print - in which case after identifying that, the warning message could be removed, and a comment be put in place instead that says that such behavior is to be expected.

For save game loading from disk and from memory buffers though, it's good to have some kind of warning at least if reading malformed input data. Though the read API there does not seem to have a way to distinguish between these two currently.